python/semantic_kernel/connectors/memory/azure_cosmos_db/utils.py

```diff
@@ -0,0 +1,213 @@
+    DistanceFunction.COSINE_SIMILARITY: "cosine",
+    DistanceFunction.DOT_PROD: "dotproduct",
+    DistanceFunction.EUCLIDEAN_DISTANCE: "euclidean",
+}
+
+
+def to_distance_function(distance_function: DistanceFunction | None) -> str:
+    """Converts the distance function to the distance function for Azure Cosmos DB NoSQL container."""
+    if distance_function in DISTANCE_FUNCTION_MAPPING:
+        return DISTANCE_FUNCTION_MAPPING[distance_function]
+
+    raise ValueError(f"Distance function '{distance_function}' is not supported by Azure Cosmos DB NoSQL container.")
+
+
+def create_default_indexing_policy(data_model_definition: VectorStoreRecordDefinition) -> dict[str, Any]:
+    """Creates a default indexing policy for the Azure Cosmos DB NoSQL container.
+
+    A default indexing policy is created based on the data model definition and has an automatic indexing policy.
+
+    Args:
+        data_model_definition (VectorStoreRecordDefinition): The definition of the data model.
+
+    Returns:
+        dict[str, Any]: The indexing policy.
+    """
+    indexing_policy = {
+        "automatic": True,
+        "includedPaths": [
+            {
+                "path": "/*",
+            }
+        ],
+        "excludedPaths": [
+            {
+                "path": '/"_etag"/?',
+            }
+        ],
+        "vectorIndexes": [],
+    }
+
+    for _, field in data_model_definition.fields.items():
+        if isinstance(field, VectorStoreRecordDataField) and (
+            not field.is_full_text_searchable and not field.is_filterable
+        ):
+            indexing_policy["excludedPaths"].append({"path": f'/"{field.name}"/*'})  # type: ignore
+
+        if isinstance(field, VectorStoreRecordVectorField):
+            indexing_policy["vectorIndexes"].append({  # type: ignore
+                "path": f'/"{field.name}"',
+                "type": to_vector_index_policy_type(field.index_kind),
+            })
+            # Exclude the vector field from the index for performance optimization.
+            indexing_policy["excludedPaths"].append({"path": f'/"{field.name}"/*'})  # type: ignore
+
+    return indexing_policy
+
+
+def create_default_vector_embedding_policy(data_model_definition: VectorStoreRecordDefinition) -> dict[str, Any]:
+    """Creates a default vector embedding policy for the Azure Cosmos DB NoSQL container.
+
+    A default vector embedding policy is created based on the data model definition.
+
+    Args:
+        data_model_definition (VectorStoreRecordDefinition): The definition of the data model.
+
+    Returns:
+        dict[str, Any]: The vector embedding policy.
+    """
+    vector_embedding_policy: dict[str, Any] = {"vectorEmbeddings": []}
+
+    for _, field in data_model_definition.fields.items():
+        if isinstance(field, VectorStoreRecordVectorField):
+            vector_embedding_policy["vectorEmbeddings"].append({
+                "path": f'/"{field.name}"',
+                "dataType": "float32",
```

Comment:
a vector field can also have Int as the datatype, which would map to`int8`: https://learn.microsoft.com/en-us/azure/cosmos-db/nosql/vector-search#vector-indexing-policies